File: taller php/funciones/funciones.php
<?php
// Operación con listas (equivalente a las tuplas de Python)

$a = array(1, 2, 3, 4, 5);
$b = array(6, 7, 8, 9, 10);
$c = array();

for ($contador = 0; $contador < count($a); $contador++) {
    $c[] = $a[$contador] * $b[$contador];
}
print_r($c);

// Funciones

// Definición y llamada de una función
function mostrar_texto() {
    echo "hola\n";
}

mostrar_texto();

// Función de multiplicación
function multiplicar() {
    $a = 7;
    $b = 9;
    echo $a * $b . "\n";
}

multiplicar();

// Ámbito de las variables (Variables dentro de una función no son accesibles fuera)
function multiplicar_ambito() {
    global $a, $b;  // Se usa 'global' para acceder a las variables globales dentro de la función
    echo $a * $b . "\n";
}

$a = 7;
$b = 9;
multiplicar_ambito();

// Funciones con valor de retorno
function multiplicar_return() {
    $a = 7;
    $b = 9;
    return $a * $b;
}

$resultado = multiplicar_return();
echo $resultado + 5 . "\n";

// Función que valida un dato
function validar_dato() {
    global $a;  // Acceder a la variable global 'a'
    if ($a == 7) {
        return true;
    } else {
        return false;
    }
}

$a = 9;
$dato = validar_dato();
echo $dato . "\n";
?>

File: GO/funciones.go
package main

import "fmt"

// Función para multiplicar los elementos de dos slices
func multiplicarSlices(a, b []int) []int {
	var c []int
	for i := 0; i < len(a); i++ {
		c = append(c, a[i]*b[i])
	}
	return c
}

func main() {
	// Tuplas en Python son similares a slices en Go
	a := []int{1, 2, 3, 4, 5}
	b := []int{6, 7, 8, 9, 10}

	// Multiplicamos los elementos de los slices
	c := multiplicarSlices(a, b)
	fmt.Println(c)

	// Funciones en Go

	// Definición y llamada de función
	mostrarTexto()

	// Función de multiplicación
	multiplicar()

	// Ámbito de las variables
	a1 := 7
	b1 := 9
	multiplicarConVariables(a1, b1)

	// Función con return
	resultado := multiplicarReturn()
	fmt.Println(resultado + 5)

	// Validar dato
	a2 := 9
	dato := validarDato(a2)
	fmt.Println(dato)
}

// Función que imprime un texto
func mostrarTexto() {
	fmt.Println("hola")
}

// Función que multiplica dos números y los imprime
func multiplicar() {
	a := 7
	b := 9
	fmt.Println(a * b)
}

// Función que usa variables externas
func multiplicarConVariables(a, b int) {
	fmt.Println(a * b)
}

// Función que devuelve el resultado de la multiplicación
func multiplicarReturn() int {
	a := 7
	b := 9
	return a * b
}

// Función que valida un dato
func validarDato(a int) bool {
	if a == 7 {
		return true
	}
	return false
}
